### ISSUE TITLE
Ability to disable tab completion based on language type

### ISSUE THREAD

**Issue by zbarbuto** [continuedev/continue#1718]
### Validations

- [ ] I believe this is a way to improve. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) that requests the same enhancement

### Problem

Currently, you can disable tab completions for particular file extensions e.g. 

```ts
  config.tabAutocompleteOptions = {
    useOtherFiles: true,
    disableInFiles: ['.env', '*.env', '*.md'],
  };
```

However, it would also be useful to disable tab completions based on the currently set language mode. For example, if you create a new file and set the language mode to "markdown" you still get completions:

<img width="603" alt="image" src="https://github.com/user-attachments/assets/4961a5c4-f92c-483d-8124-49924b50b2b0">

### Solution

The *VSCode way™* to do this would be to use language-override settings in a user's `settings.json` 

For example I might have something like:

```
  "[markdown]": {
    "continue.enableTabAutocomplete": false,
     // Any of my other settings ....
    "editor.wordWrap": "off",
  },
  "[html]": {
    // ...
  },
  // ...
```

However, continue settings don't seem to work with this feature:

<img width="372" alt="image" src="https://github.com/user-attachments/assets/5e247630-7270-477c-8d7a-5f6488966e91">

<img width="690" alt="image" src="https://github.com/user-attachments/assets/38fcb76a-584f-4213-bbb1-76863cc74059">

**Comment by sestinj** [comment]
@zbarbuto this is possible with `tabAutocompleteOptions.disableInFiles`: https://docs.continue.dev/walkthroughs/tab-autocomplete#tabautocompleteoptions

You can set an array of glob patterns that you'd like to ignore, for example ["*.md"]

**Comment by zbarbuto** [comment]
Yes, my point is that this setting doesn't work with files that are not currently saved but have a particular language mode enabled.

**Comment by sestinj** [comment]
Yeah sorry I totally skimmed over your issue before sending that : )

**Comment by sestinj** [comment]
This is a nice idea—certainly annoying to be getting unwanted completions in empty files

**Comment by zbarbuto** [comment]
Yes, it's particularly annoying in contexts where you might not necessarily need to save the file. For example a scratchpad for taking notes (a TODO-list for the next couple of hours) or extensions like [REST client](https://marketplace.visualstudio.com/items?itemName=humao.rest-client)

**Comment by zbarbuto** [comment]
I'm not much of an extension developer, but looking at some other extensions that do support this, in VSCode at least might be as simple as adding the `"scope"` property to the existing config:

See [here](https://github.com/stkb/Rewrap/blob/stable/vscode/package.json#L92) for an example.

**Comment by wilsli** [comment]
Have you tried this Continue configuration?
```json
"tabAutocompleteOptions": {
    "disableInFiles": [
      ".md"
    ]
 }
```

**Comment by zbarbuto** [comment]
Yes. I mentioned this in the issue description.

> Currently, you can disable tab completions for particular file extensions e.g.
> ```
>  config.tabAutocompleteOptions = {
>    useOtherFiles: true,
>    disableInFiles: ['.env', '*.env', '*.md'],
>  };
> ```

**Comment by zbarbuto** [comment]
Still a feature that I would like.

**Comment by plashenkov** [comment]
That's <censured> I cannot disable autocompletion for files which are unsaved, my personal notes, etc. The AI ​​intervenes and puts in its weird thoughts, and it can't be turned off

**Comment by RomneyDa** [comment]
@zbarbuto you can use `disableInFiles` in the Settings Page! comma-separated glob patterns.

**Comment by zbarbuto** [comment]
Still a feature that I would like.

**Comment by zbarbuto** [comment]
Still a feature that I would like.

_(Sorry, I'm not trying to be annoying, I'm providing activity for the bot)_

**Comment by plashenkov** [comment]
> Still a feature that I would like.

Agreed.
